### PR TITLE
UI for Email Configurations

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

Adds the necessary UI pages to support email configurations feature.
* index (with search and pagination)
* new
* edit (with delete button)

We need a proper design and microcopy for this but so far it looks like:
Screenshot 2022-01-28 at 09 12 12
Screenshot 2022-01-28 at 09 11 22
Screenshot 2022-01-28 at 17 55 32
Screenshot 2022-01-28 at 17 55 57
Screenshot 2022-01-28 at 17 56 19

**Which issue(s) this PR fixes** 

[THREESCALE-8087: Improve UX for email configuration page](https://issues.redhat.com/browse/THREESCALE-8087)

### CODE REVIEW

**Comment by akostadinov** [comment]
parent PR #2855

File: app/controllers/provider/admin/account/email_configurations_controller.rb [hunk 1]
```diff
@@ -47,14 +53,22 @@ def destroy
 
   protected
 
+  alias account current_account
+
   def configuration_params
     # For the time being we only want to support user name and password override
-    params.permit(email_configuration: %i[email user_name password])
+    params.permit(email_configuration: %i[email user_name password]).fetch(:email_configuration)
```
Comment (josemigallas) [inline app/controllers/provider/admin/account/email_configurations_controller.rb:60]: try quire(:email_configuration)

File: spec/javascripts/EmailConfigurations/components/EmailConfigurationForm.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+describe('when some fields are returned by the server', () => {
+  const emailConfiguration = {
+    id: 0,
+    email: 'hello@ollivanders.co.uk',
+    userName: 'ollivanders_wands',
+    password: '123456'
+  }
+
+  it('should fill the fields already', () => {
```
Comment (akostadinov) [inline spec/javascripts/EmailConfigurations/components/EmailConfigurationForm.spec.jsx:70]:
```suggestion
  it('should populate the fields with the values', () => {
```

File: test/functional/provider/admin/account/email_configurations_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    get edit_provider_admin_account_email_configuration_path(email_configuration)
+    assert_response :success
+  end
+
+  test '#update' do
+    email_configuration = FactoryBot.create(:email_configuration)
+    put provider_admin_account_email_configuration(email_configuration), params: email_configurations_params
+    assert_response :success
+
+    # TODO
```
Comment (akostadinov) [inline test/functional/provider/admin/account/email_configurations_controller_test.rb:34]:
```suggestion
    email_configuration.reload
    assert_equals "something", email_configuration.username
```

File: features/step_definitions/provider/admin/account/email_configurations_steps.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+Given "I have enough email configs to fill many pages" do
+  25.times do |i|
+    FactoryBot.create(:email_configuration, account: master_account,
+                                            email: "foo#{i}@example.com",
+                                            user_name: "user_#{i}",
+                                            password: "password#{i}")
+  end
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply features/step_definitions/provider/admin/account/email_configurations_steps.rb:9]: Are email configurations supposed to share the same email? For these tests, I need to populate the table with 25 rows.

[the remark this answers is not in the document]
Comment (akostadinov) [reply features/step_definitions/provider/admin/account/email_configurations_steps.rb:9]: Factory bot generates different emails. In fact the DB table has a unique insex on emails so it is not possible to have duplicates.

[the remark this answers is not in the document]
Comment (josemigallas) [reply features/step_definitions/provider/admin/account/email_configurations_steps.rb]: Gotcha!

**Comment by akostadinov** [comment]
squashed and merged into #2855, thank you!

**Pull request closed without merging**